test/static-calls.js

```diff
@@ -164,8 +165,11 @@ contract("Static Calls", (accounts) => {
       assert.isTrue(success, "enableERC1155TokenReceiver failed");
       await checkStaticCalls({ _wallet: oldWallet, _supportERC1155: true });
     });
-    it("supports ERC1155TokenReceiver by default (new wallet)", async () => {
-      await checkStaticCalls({ _wallet: wallet, _supportERC1155: true });
+
+    it("requires less than 10000 gas to call supportsInterface()", async () => {
+      // this call will fail if supportsInterface() consumes more than 10000 gas
+      const txReceipt = await (await ERC165Tester.new()).testERC165Gas(wallet.address, "0x4e2312e0");
+      console.log(`supportsInterface() costs less than ${txReceipt.logs[0].args._gas.toString()} gas`);
```

Comment:
Great, just the messaging is a bit off > supportsInterface() costs less than x gas -> supportsInterface() costs x gas

Reply:
It's actually an upper bound. The `gasleft()` adds a bit of gas that makes the amount emitted in the event slightly higher than the real cost (the amount logged is 8876 but you can see that the method still succeeds when called with only 7500 gas).